### ISSUE TITLE
gofumpt result through golangci-lint does not match standalone gofumpt

### ISSUE THREAD

**Issue by psaikko** [golangci/golangci-lint#3457]
### Welcome

- [X] Yes, I'm using a binary release within 2 latest major releases. Only such installations are supported.
- [X] Yes, I've searched similar issues on GitHub and didn't find any.
- [X] Yes, I've included all information below (version, config, etc.).
- [X] Yes, I've tried with the standalone linter if available (e.g., gocritic, go vet, etc.). (https://golangci-lint.run/usage/linters/)

### Description of the problem

Running gofumpt through golangci-lint produces different results than running standalone gofumpt.

Specifically, I get pre-v0.3.0 gofumpt import order from golangci-lint, despite having latest versions of both.

Output for the minimal code example:

```console
$ gofumpt --version
v0.4.0 (go1.19)
$ golangci-lint --version
golangci-lint has version v1.50.1 built from (unknown, mod sum: "h1:C829clMcZXEORakZlwpk7M4iDw2XiwxxKaG504SZ9zY=") on (unknown)
$ gofumpt -d a/a.go
$ golangci-lint run a/a.go --disable-all --enable gofumpt
a/a.go:5: File is not `gofumpt`-ed (gofumpt)
```

Standalone gofumpt produces no diff, but through golangci-lint the formatting check fails. Golangci-lint should be on gofumpt v0.4.0 as of v0.50.0.

### Version of golangci-lint

<details>

```console
$ golangci-lint --version
golangci-lint has version v1.50.1 built from (unknown, mod sum: "h1:C829clMcZXEORakZlwpk7M4iDw2XiwxxKaG504SZ9zY=") on (unknown)
```

</details>

### Configuration file

<details>

```console
$ cat .golangci.yml
cat: .golangci.yml: No such file or directory
```

</details>

### Go environment

<details>

```console
$ go version && go env
go version go1.19 linux/amd64
GO111MODULE=""
GOARCH="amd64"
GOBIN=""
GOCACHE="/home/psaikko/.cache/go-build"
GOENV="/home/psaikko/.config/go/env"
GOEXE=""
GOEXPERIMENT=""
GOFLAGS=""
GOHOSTARCH="amd64"
GOHOSTOS="linux"
GOINSECURE=""
GOMODCACHE="/home/psaikko/go/pkg/mod"
GONOPROXY=""
GONOSUMDB=""
GOOS="linux"
GOPATH="/home/psaikko/go"
GOPRIVATE=""
GOPROXY="https://proxy.golang.org,direct"
GOROOT="/usr/local/go"
GOSUMDB="sum.golang.org"
GOTMPDIR=""
GOTOOLDIR="/usr/local/go/pkg/tool/linux_amd64"
GOVCS=""
GOVERSION="go1.19"
GCCGO="gccgo"
GOAMD64="v1"
AR="ar"
CC="gcc"
CXX="g++"
CGO_ENABLED="1"
GOMOD="/home/psaikko/gofumpt-test2/go.mod"
GOWORK=""
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -m64 -pthread -Wl,--no-gc-sections -fmessage-length=0 -fdebug-prefix-map=/tmp/go-build3690182168=/tmp/go-build -gno-record-gcc-switches"
```

</details>

### Verbose output of running

<details>

```console
$ golangci-lint cache clean
$ golangci-lint run -v a/a.go --disable-all --enable gofumpt
INFO [config_reader] Config search paths: [./ /home/psaikko/gofumpt-test2/a /home/psaikko/gofumpt-test2 /home/psaikko /home /] 
INFO [lintersdb] Active 1 linters: [gofumpt]      
INFO [loader] Go packages loading at mode 7 (files|compiled_files|name) took 12.148589ms 
INFO [runner/filename_unadjuster] Pre-built 0 adjustments in 88.445µs 
INFO [linters_context/goanalysis] analyzers took 119.223µs with top 10 stages: gofumpt: 119.223µs 
INFO [runner] Processors filtering stat (out/in): identifier_marker: 1/1, exclude: 1/1, uniq_by_line: 1/1, max_per_file_from_linter: 1/1, max_from_linter: 1/1, source_code: 1/1, path_prettifier: 1/1, nolint: 1/1, path_prefixer: 1/1, skip_files: 1/1, diff: 1/1, max_same_issues: 1/1, path_shortener: 1/1, sort_results: 1/1, skip_dirs: 1/1, filename_unadjuster: 1/1, autogenerated_exclude: 1/1, exclude-rules: 1/1, severity-rules: 1/1, cgo: 1/1 
INFO [runner] processing took 88.396µs with stages: nolint: 22.874µs, identifier_marker: 15.353µs, autogenerated_exclude: 14.947µs, path_prettifier: 9.439µs, exclude-rules: 7.856µs, source_code: 6.197µs, skip_dirs: 3.374µs, cgo: 2.207µs, uniq_by_line: 1.741µs, max_same_issues: 1.186µs, max_from_linter: 598ns, path_shortener: 556ns, filename_unadjuster: 503ns, max_per_file_from_linter: 401ns, skip_files: 334ns, severity-rules: 232ns, diff: 222ns, exclude: 196ns, sort_results: 101ns, path_prefixer: 79ns 
INFO [runner] linters took 5.061876ms with stages: gofumpt: 4.926547ms 
a/a.go:5: File is not `gofumpt`-ed (gofumpt)

INFO File cache stats: 1 entries of total size 88B 
INFO Memory: 2 samples, avg is 28.0MB, max is 28.1MB 
INFO Execution took 20.830544ms   
```

</details>

### Code example or link to a public repository

<details>

## `a/a.go`
```go
package a

import (
	"math"

	"mre/b"
)

func Bar() float64 {
	b.Foo()
	return math.E
}
```

## `b/b.go`
```go
package b

func Foo() {

}
```

## `go.mod`
```
module mre

go 1.19
```

</details>

**Comment by ldez** [comment]
Hello,

I can't reproduce your issue:

```console
$ golangci-lint version   
golangci-lint has version 1.50.1 built from 8926a95f on 2022-10-22T10:50:47Z
$ golangci-lint run a/a.go   
$ gofumpt --version       
v0.4.0 (go1.19.4)
$ gofumpt -d a/a.go
$                                          
```

```console
$ gofumpt -d .            
diff b/b.go.orig b/b.go
--- b/b.go.orig
+++ b/b.go
@@ -1,5 +1,4 @@
 package b
 
 func Foo() {
-
 }
```

```console
$ golangci-lint run                                                    
b/b.go:4: File is not `gofumpt`-ed (gofumpt)
```

**Comment by ldez** [comment]
:thinking: seems to be related to the module name: if a use a more standard module name (as I did in my previous test) I don't have the problem.

**Comment by psaikko** [comment]
perhaps, but these module names are handled in gofumpt as expected since v0.3.0
https://github.com/mvdan/gofumpt/releases/tag/v0.3.0
https://github.com/mvdan/gofumpt/pull/199

**Comment by ldez** [comment]
I think gofumpt, as a CLI and not as Go module, auto-fill the module path.
So you can do that:
```yml
linters:
  disable-all: true
  enable:
    - gofumpt

linters-settings:
  gofumpt:
    module-path: more
```

**Comment by psaikko** [comment]
Looks like that resolves the issue for me, thanks @ldez
